File: php/features.php
<?php
declare(strict_types=1);

// Logotypes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LogotypesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LogotypesBaseFeature();
            case "test":
                return new LogotypesTestFeature();
            default:
                return new LogotypesBaseFeature();
        }
    }
}

File: py/features.py
# Logotypes SDK feature factory

from feature.base_feature import LogotypesBaseFeature
from feature.test_feature import LogotypesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LogotypesBaseFeature(),
        "test": lambda: LogotypesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
